test/correctness/simd_op_check.cpp

```diff
@@ -1767,11 +1767,24 @@ class SimdOpCheck : public SimdOpCheckTest {
                 }
 
                 // Extended pairwise integer addition
-                // TODO(https://github.com/halide/Halide/issues/5130): NOT BEING GENERATED AT TRUNK
-                // check("i16x8.extadd_pairwise_i8x16_s", ???, ???);
-                // check("i16x8.extadd_pairwise_i8x16_u", ???, ???);
-                // check("i32x4.extadd_pairwise_i16x8_s", ???, ???);
-                // check("i32x4.extadd_pairwise_i16x8_u", ???, ???);
+                for (int f : {2, 4}) {
+                    RDom r(0, f);
+
+                    // A summation reduction that starts at something
+                    // non-trivial, to avoid llvm simplifying accumulating
+                    // widening summations into just widening summations.
+                    auto sum_ = [&](Expr e) {
+                        Func f;
+                        f(x) = cast(e.type(), 123);
+                        f(x) += e;
+                        return f(x);
+                    };
+
+                    check("i16x8.extadd_pairwise_i8x16_s", 8 * w, sum_(i16(in_i8(f * x + r))));
+                    check("i16x8.extadd_pairwise_i8x16_u", 8 * w, sum_(u16(in_u8(f * x + r))));
```

Comment:
I think we should also check for signed widening from unsigned operands, e.g.: https://github.com/halide/Halide/blob/master/src/CodeGen_ARM.cpp#L1139

Reply:
There isn't a wasm instruction that does this -- just i8->i16 and u8->u16 (and wider variants). 

I guess we could just use the u8->u16 op in this case and cast the result to i16, on the assumption that the u16 result will always have a zero sign bit for 128-bit vectors....

Reply:
Yes, that's all the ARM and other targets are doing anyways. I'm not actually sure you need to change the implementation at all, it might just work (and the ARM version might be redundant too), but I think we should test for it.